design-editor/src/panel/preview/preview-element.js

```diff
@@ -172,9 +172,18 @@ class Preview extends DressElement {
 		const relativePathToFile = pathUtils.joinPaths(location, 'temporary-preview.html'),
 			writeFile = promisify(fs.writeFile);
 		this.fsPath = pathUtils.createProjectPath(relativePathToFile);
-		return writeFile(this.fsPath, contents).then(() => {
-			return pathUtils.createProjectPath(relativePathToFile, true);
-		}).catch((err) => {throw err;});
+
+		const parsedContents = this._parseBeforePreview(contents);
```

Comment:
Is there any obstacle to add this const to the beginning of the document?

Reply:
no :) I've moved it there